File: app/src/main/java/com/example/assignment/TreasureDetailsActivity.java
package com.example.assignment;

import android.os.Bundle;
import android.util.Log;
import android.view.GestureDetector;
import android.view.MotionEvent;
import android.view.View;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;
import java.util.List;



public class TreasureDetailsActivity extends AppCompatActivity {

    private TreasureDatabase database;
    private DatabaseDao databaseDao;

    private List<Treasure> treasureList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_treasure_details);

        // Initialize the database and dao
        database = TreasureDatabase.getInstance(getApplicationContext());
        databaseDao = database.databaseDao();

        // Corrected: Use a List<Treasure> for the adapter
        treasureList = databaseDao.getAllTreasures();
        List<Treasure> treasureList = databaseDao.getAllTreasures();
        for (Treasure retrievedTreasure : treasureList) {
            Log.d("TreasureData", "Name: " + retrievedTreasure.getName() +
                    ", Latitude: " + retrievedTreasure.getLatitude() +
                    ", Longitude: " + retrievedTreasure.getLongitude());
        }

        // Fetch the selected treasure details from your data source or map markers
        String treasureName = getIntent().getStringExtra("treasureName");
        String cluesText = getIntent().getStringExtra("cluesText");
        String descriptionText = getIntent().getStringExtra("descriptionText");

        // Set the retrieved details to respective TextViews or UI elements
        TextView nameTextView = findViewById(R.id.treasure_name);
        nameTextView.setText(treasureName);

        TextView cluesTextView = findViewById(R.id.clues_text);
        cluesTextView.setText(cluesText);

        TextView descriptionTextView = findViewById(R.id.description_text);
        descriptionTextView.setText(descriptionText);

        // Implement gesture listeners for navigation (e.g., back to map or next screen)
        // For example, to navigate back when a user swipes
        GestureDetector gestureDetector = new GestureDetector(this, new GestureDetector.SimpleOnGestureListener() {
            @Override
            public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX, float velocityY) {
                float deltaX = e2.getX() - e1.getX();
                if (deltaX > 0) {
                    // User swiped from left to right (navigate back)
                    onBackPressed();
                    return true;
                }
                return super.onFling(e1, e2, velocityX, velocityY);
            }
        });

        View.OnTouchListener gestureListener = (v, event) -> gestureDetector.onTouchEvent(event);

        findViewById(R.id.details_layout).setOnTouchListener(gestureListener);
    }
}

File: app/src/main/java/com/example/assignment/TreasureDetailsActivity.kt
package com.example.assignment

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.GestureDetector
import android.view.MotionEvent
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class TreasureDetailsActivity : AppCompatActivity() {

    @SuppressLint("ClickableViewAccessibility")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_treasure_details)

        // Fetch the selected treasure details from your data source or map markers
        val treasureName = intent.getStringExtra("treasureName")
        val cluesText = intent.getStringExtra("cluesText")
        val descriptionText = intent.getStringExtra("descriptionText")

        // Set the retrieved details to respective TextViews or UI elements
        val nameTextView = findViewById<TextView>(R.id.treasure_name)
        nameTextView.text = treasureName

        val cluesTextView = findViewById<TextView>(R.id.clues_text)
        cluesTextView.text = cluesText

        val descriptionTextView = findViewById<TextView>(R.id.description_text)
        descriptionTextView.text = descriptionText

        // Implement gesture listeners for navigation (e.g., back to map or next screen)
        // For example, to navigate back when a user swipes
        val gestureDetector = GestureDetector(this, object : GestureDetector.SimpleOnGestureListener() {
            override fun onFling(
                e1: MotionEvent?,
                e2: MotionEvent,
                velocityX: Float,
                velocityY: Float
            ): Boolean {
                val deltaX: Float = e2.x - e1.x
                if (deltaX > 0) {
                    // User swiped from left to right (navigate back)
                    onBackPressed()
                    return true
                }
                return super.onFling(e1, e2, velocityX, velocityY)
            }
        })

        val gestureListener = View.OnTouchListener { _, event ->
            gestureDetector.onTouchEvent(event)
        }

        findViewById<View>(R.id.details_layout).setOnTouchListener(gestureListener)
    }
}
